selfservice/strategy/code/strategy_login.go

```diff
@@ -136,18 +143,14 @@ func (s *Strategy) findIdentityByIdentifier(ctx context.Context, identifier stri
 	return id, false, nil
 }
 
-func (s *Strategy) Login(w http.ResponseWriter, r *http.Request, f *login.Flow, _ uuid.UUID) (_ *identity.Identity, err error) {
+func (s *Strategy) Login(w http.ResponseWriter, r *http.Request, f *login.Flow, sess *session.Session) (_ *identity.Identity, err error) {
 	ctx, span := s.deps.Tracer(r.Context()).Tracer().Start(r.Context(), "selfservice.strategy.code.strategy.Login")
 	defer otelx.End(span, &err)
 
 	if err := flow.MethodEnabledAndAllowedFromRequest(r, f.GetFlowName(), s.ID().String(), s.deps); err != nil {
 		return nil, err
 	}
 
-	if err := login.CheckAAL(f, identity.AuthenticatorAssuranceLevel1); err != nil {
```

Comment:
I don't think this should be removed, but rather conditional on whether the code method can be used for 1fa or 2fa. Otherwise I think that we'll have problems with incorrect "you wanted 2FA but you have only 1FA" error handling

Reply:
Reverted.